sdk/tests/core/test_az_http.c

```diff
@@ -560,22 +560,87 @@ static void test_http_response(void** state)
     }
   }
 
-  az_span response_span = AZ_SPAN_FROM_STR( //
-      "HTTP/1.1 200 Ok\r\n"
-      "Content-Type: text/html; charset=UTF-8\r\n"
-      "\r\n"
-      // body
-      EXAMPLE_BODY);
-
-  az_http_response response = { 0 };
-  az_result const result = az_http_response_init(&response, response_span);
-  assert_true(result == AZ_OK);
-
-  // add parsing here
+  // Processing valid response
   {
-    az_span body;
-    TEST_EXPECT_SUCCESS(az_http_response_get_body(&response, &body));
-    assert_true(az_span_is_content_equal(body, AZ_SPAN_FROM_STR(EXAMPLE_BODY)));
+    az_span response_span = AZ_SPAN_FROM_STR( //
+        "HTTP/1.1 200 Ok\r\n"
+        "Content-Type: text/html; charset=UTF-8\r\n"
+        "\r\n"
+        // body
+        EXAMPLE_BODY);
+
+    az_http_response response = { 0 };
+    az_result const result = az_http_response_init(&response, response_span);
+    assert_true(result == AZ_OK);
+
+    // get body directly
+    {
+      az_span body;
+      TEST_EXPECT_SUCCESS(az_http_response_get_body(&response, &body));
+      assert_true(az_span_is_content_equal(body, AZ_SPAN_FROM_STR(EXAMPLE_BODY)));
+    }
+  }
+
+  // Bad response. Will get first part right and will block reading beyond the digits
+  {
+    az_span response_span = AZ_SPAN_FROM_STR( //
+        "HTTP/");
+
+    az_http_response response = { 0 };
+    az_result const result = az_http_response_init(&response, response_span);
+    assert_true(result == AZ_OK);
```

Comment:
Same here and elsewhere:
Missed this and other instances of `assert_true` -> `assert_int_equal`

I didn't call out all the lines for such instance. Maybe we should file an issue to fix up all `az_result` asserts for tests to use `assert_int_equal`